lib/database/migrations/20240424092020-add-qc-flag-effective-period-columns.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { timestampToMysql } = require('../../server/utilities/timestampToMysql');
+
+/**
+ * Discard effective periods of flags belonging to given scope
+ * @param {QueryInterface} queryInterface sequelize interface
+ * @param {Period} period period which discards other effective periods
+ * @param {object} scope scope of QC flags
+ * @param {object} scope.runNumber run number
+ * @param {object} scope.dplDetectorId dpl detector id
+ * @param {object} [scope.dataPassId] data pass id - exclusive with simulationPassId
+ * @param {object} [scope.simulationPassId] simulation pass id - exclusive with dataPassId
+ * @param {object} scope.createdAt creation timestamp of discarding flag,
+ * only effective periods of QC flags created before this timestamp will be updated
+ * @return {Promise<void>} promise
+ */
+const _updateEffectivePeriodForScopeAndPeriod = async (
+    queryInterface,
+    { from: intersectingPeriodFrom, to: intersectingPeriodTo },
+    { runNumber, dplDetectorId, dataPassId, simulationPassId, createdAt },
+    { transaction },
+) => {
+    let joinClause =
+    `
+        INNER JOIN quality_control_flags as qcf
+            ON qcfep.flag_id = qcf.id
+                AND qcf.run_number = ${runNumber} 
+                AND qcf.dpl_detector_id = ${dplDetectorId}
+                AND unix_timestamp(qcf.created_at) * 1000 < ${createdAt}
```

Comment:
You mix `timestampToMysql` and `unix_timestamp`
I think using simply `timestampToMysql` would make it more clear (no need for *1000), I would advice to use always this.
You can also extract a variable, for example I would keep `period` from param, then do
```js
intersectingPeriodFrom = timestampToMysql(period.from);
intersectingPeriodTo = timestampToMysql(period.to);
```

Reply:
applied